src/internal_events/pulsar.rs

```diff
@@ -56,3 +58,101 @@ impl<F: std::fmt::Display> InternalEvent for PulsarPropertyExtractionError<F> {
         );
     }
 }
+
+pub enum PulsarErrorEventType {
+    ReadError,
+    AckError,
+    NAckError,
+}
+
+pub struct PulsarErrorEventData {
+    pub msg: String,
+    pub error_type:PulsarErrorEventType,
+}
+
+registered_event!(
+    PulsarErrorEvent => {
+        ack_errors_count: Histogram = register_histogram!(
+            "component_errors_count",
+            "error_code" => "acknowledge_message",
+            "error_type" => error_type::ACKNOWLEDGMENT_FAILED,
+            "stage" => error_stage::RECEIVING,
+        ),
+        ack_errors: Counter = register_counter!(
+            "component_errors_total",
+            "error_code" => "acknowledge_message",
+            "error_type" => error_type::ACKNOWLEDGMENT_FAILED,
+            "stage" => error_stage::RECEIVING,
+        ),
+
+        nack_errors_count: Histogram = register_histogram!(
+            "component_errors_count",
+            "error_code" => "negative_acknowledge_message",
+            "error_type" => error_type::ACKNOWLEDGMENT_FAILED,
+            "stage" => error_stage::RECEIVING,
+        ),
+        nack_errors: Counter = register_counter!(
+            "component_errors_total",
+            "error_code" => "negative_acknowledge_message",
+            "error_type" => error_type::ACKNOWLEDGMENT_FAILED,
+            "stage" => error_stage::RECEIVING,
+        ),
+
+        read_errors_count: Histogram = register_histogram!(
+            "component_errors_count",
+            "error_code" => "reading_message",
+            "error_type" => error_type::READER_FAILED,
+            "stage" => error_stage::RECEIVING,
+        ),
+        read_errors: Counter = register_counter!(
+            "component_errors_total",
+            "error_code" => "reading_message",
+            "error_type" => error_type::READER_FAILED,
+            "stage" => error_stage::RECEIVING,
+        ),
+    }
+
+    fn emit(&self,error:PulsarErrorEventData) {
+        match error.error_type{
+            PulsarErrorEventType::ReadError=>{
```

Comment:
Just eyeballing this, I think our format check should fail. Make sure to run `cargo fmt`.